### PR TITLE
BLOGS-1415 Implement "publisher" event tracking for Blog Index page

### PR DESCRIPTION
**Pull request by Tasiobg**
https://jira.dev.bbc.co.uk/browse/BLOGS-1415

I tested it and seems fine but I have to open the PR in a bit of a rush becuase I'm going on holidays, please feel free to do any changes and merge when ready

### CODE REVIEW

File: src/Ds/Post/Author/author.html.twig [hunk 1]
```diff
@@ -1,4 +1,6 @@
-<a data-section="author" class="block {{- author.getOption('is_slimline') ? '' : ' islet--vertical' -}}" href="{{ path('author_show', {blogId: author.getBlogId(), guid: author.getGuid()}) }}" data-bbc-container="{{ author.getBlogId() }}" data-bbc-title="post-author">
+<a data-section="author" class="block {{- author.getOption('is_slimline') ? '' : ' islet--vertical' -}}"
+   href="{{ path('author_show', {blogId: author.getBlogId(), guid: author.getGuid()}) }}" data-bbc-container="{{ author.getBlogId() }}"
+   data-bbc-title="{{ author.getOption('is_featured_post') ? '' : 'featured-' }}post-author">
```
Comment (benolot) [inline src/Ds/Post/Author/author.html.twig]: Isn't this the wrong way around? I thought it was evaulation ? true : false
```suggestion
   data-bbc-title="{{ author.getOption('is_featured_post') ? 'featured-' : '' }}post-author">
```

Comment (Tasiobg) [reply src/Ds/Post/Author/author.html.twig]: could be, I think I refactored the code to make it shorted and I may have miss this

File: src/Ds/Post/PostSummary/post_summary.html.twig [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <div>
     <h{{ post_summary.getOption('h_tag') }} class="{{ post_summary.getOption('h_class') }} no-margin">
         <a
-            data-bbc-container="{{ post_summary.getBlogId() }}" data-bbc-title="post-title"
+            data-bbc-title="{% if post_summary.isFeaturedPost() %}featured-{% endif %}post-title" data-bbc-container="{{ post_summary.getBlogId() }}"
```
Comment (benolot) [inline src/Ds/Post/PostSummary/post_summary.html.twig]:
```suggestion
            data-bbc-title="{{  post_summary.isFeaturedPost() ? 'featured-' : '' }}post-title" data-bbc-container="{{ post_summary.getBlogId() }}"
```
Keeping it in line with the other ones like the one above

File: src/Ds/Post/PostSummary/post_summary.html.twig [hunk 3]
```diff
@@ -27,13 +27,14 @@
                 <p>{{ post_summary.getPost().getShortSynopsis() }}</p>
             {% endif %}
             <p>
-                <a  data-link="read-more"
-                    data-bbc-container="{{ post_summary.getBlogId() }}" data-bbc-title="post-read-more"
+                <a
+                    data-bbc-title="{% if post_summary.isFeaturedPost() %}featured-{% endif %}post-read-more" data-bbc-container="{{ post_summary.getBlogId() }}"
```
Comment (benolot) [inline src/Ds/Post/PostSummary/post_summary.html.twig]:
```suggestion
                    data-bbc-title="{{  post_summary.isFeaturedPost() ? 'featured-' : ''}}post-read-more" data-bbc-container="{{ post_summary.getBlogId() }}"
```

File: src/Ds/Post/PostPreview/post_preview.html.twig [hunk 4]
```diff
@@ -1,6 +1,7 @@
 <div>
     <h3 class="beta">
-        <a href="{{ path('post', {blogId: post_preview.getBlogId(), guid: post_preview.getPost().getGuid()}) }}">
+        <a data-bbc-title="post-title" data-bbc-container="{{ post_preview.getBlogId() }}"
```
Comment (benolot) [inline src/Ds/Post/PostPreview/post_preview.html.twig]: Featured isn't needed here?

**Pull request merged**